### ISSUE TITLE
When the Delete Credential required action is set to false an authentication application cannot be removed from the account UI

### ISSUE THREAD

**Issue by transducer** [keycloak/keycloak#30204]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

account/ui

### Describe the bug

When the required action Delete Credential is set to false in the UI at /admin/master/console/#/master/authentication/required-actions 
<img width="1437" alt="Screenshot 2024-06-06 at 11 24 27" src="https://github.com/keycloak/keycloak/assets/5002921/55b5bbda-7849-48d0-8adc-98badff2f4ca">

the Delete button in the UI at /realms/master/account/#/account-security/signing-in

<img width="1428" alt="Screenshot 2024-06-06 at 11 20 21" src="https://github.com/keycloak/keycloak/assets/5002921/0ad990c1-0ab1-4e53-a99a-f2680978f48d">

stops working. You can click on it and the page refreshes, but the device is not removed.

Note that despite the migration code for 24.0.3 setting the new setting to true [here](https://github.com/keycloak/keycloak/blob/3e243d77ec84663c930301b2621aa311860a4f2b/model/storage-private/src/main/java/org/keycloak/migration/migrators/MigrateTo24_0_3.java#L54) and [here](https://github.com/keycloak/keycloak/blob/24.0.3/server-spi-private/src/main/java/org/keycloak/models/utils/DefaultRequiredActions.java#L216), the setting Delete Credential was set to false for us after a migration from 23.0.0. This means that after migration the deletion of the device from the account UI stopped working.

### Version

24.0.5

### Regression

- [ ] The issue is a regression

### Expected behavior

The button works and the page navigates to /realms/master/login-actions/required-action?execution=delete_credential&client_id=account-console&tab_id=kXy2mtc9h1A

<img width="1129" alt="Screenshot 2024-06-06 at 11 30 04" src="https://github.com/keycloak/keycloak/assets/5002921/5f28a316-71b7-41fc-a000-fde169eede3d">

### Actual behavior

The page refreshes, but device is not removed.

### How to Reproduce?

1. Migrate to version 24.0.5 from a version before 24.0.3 OR start Keycloak using
    ```shell
    docker run -p 8081:8080 \
      -e KEYCLOAK_ADMIN=admin
      -e KEYCLOAK_ADMIN_PASSWORD=admin \
      quay.io/keycloak/keycloak:24.0.5 start-dev
    ```
    and login with user admin with password admin and navigate to /admin/master/console/#/master/authentication/required-actions and set Disable Credential to Off.
1. Navigate to the account page at /realms/master/account/#/account-security/signing-in
1. Set up an Authenticator application
1. Click Delete, and the page will refresh, but not remove the device.

**Comment by ssilvert** [comment]
This doesn't sound like a UI issue.  Moving to core teams.

**Comment by dembaca** [comment]
For the record: Same problem in RHBK 22.0.11.redhat-00001

**Comment by mposolda** [comment]
In case that `Delete credential` action is disabled, it is expected behaviour that it is not possible to delete the credentials. However I see possible issues here:

- In case that `Delete credential` action is disabled, we should hide the `Remove` button from the account console as user is not able to remove the credential (due the disabled action `delete credential`) and button does not do anything. So will be better for usability to not display this button at all IMO

- Migration does not work as expected. This might be different issue as it is expected that after the migration the `Delete Credential` action will be created and enabled.

@transducer @dembaca Could you please clarify if the action was disabled for you after the migration? Or did you manually disable it afterwards? If it was disabled after the migration (without you manually disabling it), could you please add some details about your environment (DB vendor used including version, environment (bare metal or container) etc)?

**Comment by transducer** [comment]
@mposolda the action was set to disabled after the migration, later I manually said it back to true. We are using a container with postgres (version 11). We are also using KC_CACHE ispn and have two replicas. Starting up we had to manually delete old pods in Kubernetes since the new ones were stuck finding the other instances, but this is probably unrelated. Otherwise default config.

**Comment by mposolda** [comment]
@transducer Thanks, few things to this:

1) The migration issue: I was not able to reproduce that. I did exactly:

- Started Keycloak 24.0.2 against PostgreSQL 11.5. Created some realms in that one.

- Stop the Keycloak 24.0.2.

- Started Keycloak 24.0.5 against same PostgreSQL DB. After the migration, all the realms had enabled `Delete Credential` action set in them

If this is still an issue, we will need some more details on how exactly to reproduce the migration. Besides that, it is not possible to reproduce and fix it.

2) Hiding the `delete` button in the account console when `Delete credential` action is disabled: Sent PR for that one https://github.com/keycloak/keycloak/pull/34150

**Comment by jonkoops** [comment]
@mposolda is this something we might want to backport?

**Comment by transducer** [comment]
@mposolda thank you for your efforts and the fixing of the delete button issue. I don't know an exact reproduction scenario for the migration issue, but we did start with a much earlier version of Keycloak than 24. To be precise, we started with `jboss/keycloak:8.0.1`. Anyway, since I am the only one reporting this migration issue and there's no reproducer, I think it's fine to close it. Thank you.
